File: src/vsb/graphinterfaces/Graph.java
package vsb.graphinterfaces;

import java.util.Set;

public interface Graph {
	
	void addVertex(Vertex vertex);
	void addEdge(Vertex vertexA, Vertex vertexB);
	Set<Vertex> getVertices();
	Set<Edge> getEdges();
	Set<Vertex> getVerticesBy(VertexFilter filter);
	Set<Edge> getEdgesBy(EdgeFilter filter);
	
}

File: src/vsb/graphinterfaces/Graph.scala
package vsb.graphinterfaces

trait Graph {

	def addVertex(vertex: Vertex)
	def addEdge(vertexA: Vertex, vertexB: Vertex)
	def getVertices: scala.collection.Set[Vertex]
	def getEdges: scala.collection.Set[Edge]
	def getVerticesBy(filter: VertexFilter): scala.collection.Set[Vertex]
	def getEdgesBy(filter: EdgeFilter): scala.collection.Set[Edge]
	
}
